grails-datastore-gorm-rx/src/main/groovy/org/grails/gorm/rx/events/MultiTenantEventListener.groovy

```diff
@@ -55,12 +55,13 @@ class MultiTenantEventListener implements PersistenceEventListener {
                         TenantId tenantId = entity.getTenantId();
                         if(tenantId != null) {
                             Serializable currentId = Tenants.currentId(datastoreClient.getClass());
-                            query.eq(tenantId.getName(), currentId );
+                            if (currentId != ConnectionSource.DEFAULT) {
```

Comment:
@graemerocher The reason I decided to add this check here is that in case of non shared multi-tenant database (e.g. multi-tenancy across databases) we might be using that default data source in case `tenantId` is not specified and we do not know in `Tenants.groovy` that if it's a shared or nonshared multi-tenancy. However, I am not sure about the above logic so please suggest if we need to fix this some other way.